src/tests/ftest/soak/utils.py

```diff
@@ -144,52 +145,76 @@ def reserved_file_copy(self, file, pool, container, num_bytes=None, cmd="read"):
         fscopy_cmd.run()
 
 
-def get_remote_logs(self):
+def get_remote_dir(self, source_dir, dest_dir, host_list, shared_dir=None,
+                   rm_remote=True, append=None):
     """Copy files from remote dir to local dir.
 
     Args:
         self (obj): soak obj
+        source_dir (str): Source directory to archive
+        dest_dir (str): Destinaton directory
+        host_list (list): list of hosts
 
     Raises:
         SoakTestError: if there is an error with the remote copy
 
     """
-    # copy the files from the client nodes to a shared directory
-    command = "/usr/bin/rsync -avtr --min-size=1B {0} {1}/..".format(
-        self.test_log_dir, self.sharedsoakdir)
-    result = slurm_utils.srun(
-        NodeSet.fromlist(self.hostlist_clients), command, self.srun_params)
-    if result.exit_status == 0:
-        # copy the local logs and the logs in the shared dir to avocado dir
-        for directory in [self.test_log_dir, self.sharedsoakdir]:
-            command = "/usr/bin/cp -R -p {0}/ \'{1}\'".format(
-                directory, self.outputsoakdir)
+    if shared_dir is None:
+        shared_dir = self.sharedsoaktest_dir
+    if append:
+        for host in host_list:
+            shared_dir_tmp = shared_dir + append + "{}".format(host)
+            dest_dir_tmp = dest_dir + append + "{}".format(host)
+            if not os.path.exists(shared_dir_tmp):
+                os.mkdir(shared_dir_tmp)
+            if not os.path.exists(dest_dir_tmp):
+                os.mkdir(dest_dir_tmp)
+            # copy the directory from each client node to a shared directory
+            # tagged with the hostname
+            command = "/usr/bin/rsync -avtr --min-size=1B {0} {1}/..".format(
+                source_dir, shared_dir_tmp)
             try:
-                result = run_command(command, timeout=30)
+                slurm_utils.srun(NodeSet.fromlist([host]), command, self.srun_params)
             except DaosTestError as error:
                 raise SoakTestError(
-                    "<<FAILED: job logs failed to copy>>") from error
+                    "<<FAILED: Soak remote logfiles not copied from clients>>: {}".format(host))
+            command = "/usr/bin/cp -R -p {0}/ \'{1}\'".format(shared_dir_tmp, dest_dir)
+            try:
+                run_command(command, timeout=30)
+            except DaosTestError as error:
+                raise SoakTestError("<<FAILED: job logs failed to copy>>") from error
+    else:
+        # copy the remote dir on all client nodes to a shared directory
+        command = "/usr/bin/rsync -avtr --min-size=1B {0} {1}/..".format(
+            source_dir, shared_dir)
+        try:
+            slurm_utils.srun(NodeSet.fromlist(host_list), command, self.srun_params)
+        except DaosTestError as error:
+            raise SoakTestError(
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword